### ISSUE TITLE
Deleting one/more key-value pair from the secret in AWS SM does not remove from the K8s Secrets

### ISSUE THREAD

**Issue by samof76** [external-secrets/external-secrets#315]
**Describe the solution you'd like**
Deleting one/more key-value pair from the secret in AWS SM, the sync does not remove those removed key-value pairs. Using `dataFrom`.

**What is the added value?**
The secrets generated by external secrets must be representative of what is in the AWS SM.

**Give us examples of the outcome**

Assume this is the initial secret store in the AWS SM.

```json
{
  "REDIS_PASSWORD": "rPassword",
  "REDIS_USERNAME": "rUsername",
  "FOO": "BAR",
  "BAZ": "BAN"
}
```
No if I remove the secret `FOO` and `BAZ` from the above defined secret, post the sync interval both `FOO` and `BAZ` still remain in the k8s secrets generated by external secrets.

**Observations (Constraints, Context, etc):**
NA

**Comment by spirosoik** [comment]
I can have a look what we can do.

**Comment by spirosoik** [comment]
The issue exists and I have found the cause, I will have a patch

**Comment by knelasevero** [comment]
We can close this right? @spirosoik @waylonm ? I will cut a new release when I have some time

**Comment by booleanbetrayal** [comment]
@knelasevero / @spirosoik  - We are seeing this issue on `v0.3.11`. Perhaps it was re-introduced as a regression? We are also using `dataFrom`, consolidating multiple secrets from AWS Secrets Manager.

**Comment by booleanbetrayal** [comment]
@knelasevero / @spirosoik - My Go is bad, but it looks like the `Merge*` methods in the `utils` package is possibly doing the additive only change that was fixed in #327 . Can one of you confirm?

**Comment by moolen** [comment]
I think it's this codepath that is responsible for that issue (farther up top in the file is the fix from #327):

https://github.com/external-secrets/external-secrets/blob/0b9c142a22b98c0845bc15dcfb5992d1ddc2aaa1/pkg/controllers/externalsecret/externalsecret_controller_template.go#L64-L71
